analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -1666,3 +1666,40 @@ public void Nested(object arg)
     private object WithDelegate(Func<IEnumerable<object>, object> expression) => null;
     private void WithDelegate(Func<object> expression) { }
 }
+
+public class Peach_Sharex_Project
+{
+    public void TwoVariables(object[] argArray)
+    {
+        object item;
+        if (argArray != null && argArray.Length > 1)
+        {
+            item = argArray[0];
+        }
+        else
+        {
+            item = null;
+        }
+        if (item != null)
+        {
+            argArray.ToString();    // Compliant
+        }
+    }
+
+    public void StringIsNullOrEmpty(string value)
+    {
+        if (value == null)
+        {
+            value = "";
+        }
+        var isEmpty = string.IsNullOrEmpty(value);
+        if (isEmpty)
+        {
+            value.ToString();   // Compliant
+        }
+        else
+        {
+            value.ToString();   // Compliant
```

Comment:
I don't understand this. On peach we get an FP here, right? If so I would have expected `// Noncompliant FP` here. This doesn't reproduce the FP then. Am I missing something?

Reply:
Exactly, it's a bit weird. There's probably some funky difference in CFG 😒 

I've added these UTs to be sure we don't raise on those under "normal" conditions. And to be able to catch the regressions, if something happens.

Reply:
So we are going to ignore the peach issue, right? And we wait for FP reports from users before taking any further action, right? Does it make sense to leave a comment on peach, that we were not able to reproduce it?

Reply:
I've added comments, not sure if we will ever notice them :/